### PR TITLE
lazy configure ReactAndroid gradle tasks

### PR DESCRIPTION
**Pull request by dulmandakh**
## Summary

ReactAndroid Gradle was failing mysteriously with error below, because it was trying to read values from **android** when it wasn't configured completely.

```java
extensionSupplier.get()!!.compileSdkVersion must not be null
```

or 

```java
compileSdkVersion is not specified.
```

It is happening because **buildReactNdkLib** task was created and configured eagerly. So this PR changes some tasks to be configured lazily, and reads values from **android** when ready. Also remove ANDROID_NDK variable check because android gradle plugin doing it automatically.

## Changelog

[Android] [Changed] - lazily configure ReactAndroid gradle tasks

## Test Plan

./gradlew ReactAndroid:tasks run without errors, while master throws exception.

### CODE REVIEW

**Comment by dulmandakh** [comment]
@friederbluemle @gengjiawen please review

**Comment by friederbluemle** [comment]
Awesome, it's working for me locally! 👍 Thanks @dulmandakh 🎉

File: ReactAndroid/build.gradle [hunk 1]
```diff
@@ -210,14 +210,9 @@ def findNdkBuildFullPath() {
         def ndkDir = property("ndk.path")
         return new File(ndkDir, getNdkBuildName()).getAbsolutePath()
     }
-    if (System.getenv("ANDROID_NDK") != null) {
```
Comment (gengjiawen) [inline ReactAndroid/build.gradle]: Still keep the getenv logic ?

**Comment by dulmandakh** [comment]
Android plugin is reading it and making available via android.ndkDirectory,
so I removed it.

**Comment by dulmandakh** [comment]
@friederbluemle please approve it if it worked 😉

File: ReactAndroid/build.gradle (hunk 1 above)
Comment (dulmandakh) [reply ReactAndroid/build.gradle]: Android plugin is reading it and making available via android.ndkDirectory, so I removed the duplication.

**Comment by dulmandakh** [comment]
@mdvacca any news?

**Comment by mdvacca** [comment]
trying one more time...

**Comment by mdvacca** [comment]
@dulmandakh this PR keeps failing with the following error:
* What went wrong:
Could not determine the dependencies of task ':ReactAndroid:packageReactNdkLibs'.
> Could not create task ':ReactAndroid:buildReactNdkLib'.
   > NDK is not installed

I think the change from Line 213 is affecting the build?

**Comment by friederbluemle** [comment]
@mdvacca - Which task are you running that is failing?

**Comment by dulmandakh** [comment]
@mdvacca please make sure that you have NDK installed and ANDROID_NDK environment variable points to it.

**Comment by dulmandakh** [comment]
@mdvacca the :ReactAndroid:buildReactNdkLib task works as expected if NDK is set

**Comment by dulmandakh** [comment]
@cpojer please merge

**Comment by dulmandakh** [comment]
@mdvacca @cpojer please review and merge. I'm working on other gradle improvements, and I need it.

**Comment by friederbluemle** [comment]
I just tested this again, and it (still) works. :+1:

**Comment by cpojer** [comment]
@dulmandakh CI is failing with the same error @mdvacca mentioned above. Could you fix that first?

**Comment by dulmandakh** [comment]
@cpojer sorry, fixed. I'm working to add Gradle builds to CI https://github.com/facebook/react-native/pull/26482

**Pull request closed without merging**